### DISCUSSION TITLE
Why does llama.cpp implement GGML again in the repo instead of referencing the GGML repo?

### DISCUSSION THREAD

**Discussion by Ucag** [ggml-org/llama.cpp#6381] [Q&A] [answered]
I noticed that `llama.cpp` reimplement `ggml` in the repo. Why not use the existing [ggml](https://github.com/ggerganov/ggml) repo ? Is there any difference between the two implementations? Can I use `ggml` repo to replace the implementation of `llama.cpp`?

**Comment by phymbert** [comment, marked as answer]
It is a playground for GGML as described here:
- https://github.com/ggerganov/llama.cpp?tab=readme-ov-file#description
- Sync to llama.cpp: https://github.com/ggerganov/llama.cpp/pull/6351
- Sync from llama.cpp: https://github.com/ggerganov/ggml/pull/773
